swift_scality_backend/diskfile.py

```diff
@@ -354,12 +403,39 @@ def create(self, size=None):
                      `DiskFile` class in Swift. This `create` method is
                      called externally only by the `ObjectController`
         """
-        yield DiskFileWriter(self.client_collection, self._name, self.logger)
+        yield DiskFileWriter(self._client_collection, self._name, self._logger)
 
     @utils.trace
-    def write_metadata(self, metadata):
+    def write_metadata(self, md_to_add):
         """Write a block of metadata to an object."""
-        self.client_collection.get_write_client().put_meta(self._name, metadata)
+
+        if not self._metadata:
+            md_to_write = copy.deepcopy(md_to_add)
+            md_to_write.update({
+                'df': {
+                    'name': self._name,
+                },
+                'mf': md_to_add
+            })
+            self.client_collection.get_write_client().put_meta(self._name, md_to_write)
+            return
+
+        # Keep the most recent Content-Type from either datafile or metafile metadata
+        if 'Content-Type' not in md_to_add and self._metadata is not None \
```

Comment:
That's a lot of conditions in one `if`. I'd consider rewriting as
```python
if all([
        'Content-Type' not in md_to_add,
        self._metadata is not None,
        'Content-Type' in self._metadata['mf'],
        ...]):
    md_to_add(...)
    ...
```

Reply:
Fixed